src/targets/gpu/device/include/migraphx/gpu/device/nary.hpp

```diff
@@ -259,20 +290,25 @@ void nary_standard_impl(hipStream_t stream, F f, argument result, Arguments... a
 template <class F, class... Arguments>
 void nary_impl(hipStream_t stream, F f, argument result, Arguments... args)
 {
-    bool standard = all_of({args.get_shape()...}, [](const shape& s) { return s.standard(); });
-    bool packed   = all_of({args.get_shape()...}, [](const shape& s) { return s.packed(); });
-    bool same_shapes =
-        all_of({args.get_shape()...}, [&](const shape& s) { return s == result.get_shape(); });
-    if(standard or (packed and same_shapes))
+    MIGRAPHX_TRACE_NARY_FUNCTION
+    const auto shapes   = make_array(args.get_shape()...);
+    const bool standard = all_of(shapes, [](const shape& s) { return s.standard(); });
+    const bool packed   = all_of(shapes, [](const shape& s) { return s.packed(); });
+    const bool same_shapes =
+        all_of(shapes, [&](const shape& s) { return s == result.get_shape(); });
+    const bool same_input_shapes = all_of(shapes, [&](const shape& s) { return s == shapes[0]; });
+    if((result.get_shape().standard() and standard) or (packed and same_shapes))
         nary_standard_impl(stream, f, result, args...);
+    else if(packed and same_input_shapes)
```

Comment:
do we have a test case for this?

Reply:
Yes the `test_concat_transpose` will test this path.

Reply:
I see that with concat, you're iterating through each input and calling contiguous() with a single argument. In this case, same_input_shapes will always be true since there's only one input shape. Do we have a test case where that's not true?

Reply:
For `test_concat_transpose` the `same_input_shapes` is true but `same_shapes` is not true. However, everything is packed so it will be called. If the inputs are non-packed and the output is then it will hit the final else.

Reply:
That makes sense, but I'm mainly asking if there are other operators to test that use nary() where the input shapes are not the same but the data is packed. If not, then I don't know if we need to compute another boolean just for this purpose.